File: nightly/tests/java/ExceptionFlow.java
// ExceptionFlow.java
public final class ExceptionFlow {
    static String source() {
        return "tainted";
    }

    static void sink(String s) {
        System.out.println(s);
    }

    public static void main(String[] args) {
        try {
            String data = source();
            throw new Exception(data);
        } catch (Exception e) {
            sink(e.getMessage());
        }
    }
}

File: nightly/tests/python/exceptionflow.py
# Taint carried as an exception payload across a raise/except boundary. The
# tainted value is wrapped in a ValueError, raised, caught, and recovered from
# err.args before reaching sink(). Exercises taint flow through exception objects.
def main():
    try:
        raise ValueError(source())
    except ValueError as err:
        leaked = err.args[0]
        sink(leaked)
